File: C++/4-Praktikum/4/AddExpression.cpp
// Nama         : Naufal Dean Anugrah
// NIM          : 13518123
// Tanggal      : 13 Februari 2020 
// Topik        : Inheritance, Polymorphism, Abstract Base Class

#include "AddExpression.hpp"

AddExpression::AddExpression(Expression * T1, Expression * T2) : BinaryExpression(T1, T2) {}

int AddExpression::solve() {
	return x->solve() + y->solve();
}

File: Java/7-Tutorial/2/AddExpression.java
// Nama		: Naufal Dean Anugrah
// NIM		: 13518123
// Tanggal	: 12 Maret 2020
// Topik 	: Pengenalan java

class AddExpression implements Expression {
	protected Expression x;
	protected Expression y;

	public AddExpression(Expression x, Expression y) {
		this.x = x;
		this.y = y;
	}

	public int solve() {
		return this.x.solve() + this.y.solve();
	}
}